### DISCUSSION TITLE
Switch to readthedocs?

### DISCUSSION THREAD

**Discussion by baseplate-admin** [emmyarty/dxsvelte#5] [Ideas]
Hi there, 

I think it would be better if we switched to readthedocs.org for documentation.

**Comment by emmyarty** [comment]
I think a rework of the documentation itself wouldn't be a bad idea for better clarity etc, though it's not the highest of priorities at the moment for me at the minute. The documentation currently doubles up as my 'playground project' where I test new features etc. and it's quite a convenient way to publicly demo a basic example of what DxSvelte can do.

And then there's this 😋

![image](https://user-images.githubusercontent.com/70502936/232514030-fda495d3-0311-4690-b559-fd629cd69e12.png)

With all due love to RTD of course, they host massive volumes of free information and have a highly structured way of managing content, but with how little complexity there is to actually using DxSvelte the benefit doesn't justify the time cost for me right now.

That said, if you find anything in particular about the documentation unclear let me know and I'll happily address it!

**Reply by baseplate-admin** [reply]
> I think a rework of the documentation itself wouldn't be a bad idea for better clarity etc, though it's not the highest of priorities at the moment for me at the minute. The documentation currently doubles up as my 'playground project' where I test new features etc. and it's quite a convenient way to publicly demo a basic example of what DxSvelte can do.

Fair enough..

> And then there's this yum

I am genuinely impressed this is achievable... Well done mate

> That said, if you find anything in particular about the documentation unclear let me know and I'll happily address it!

I was thinking of removing the milestone information from `README.md` file :) ( or maybe moving it downwards? )

---

All in all thanks for this awesome library. I have a project in incubation that i am eager to rewrite in this library

**Reply by emmyarty** [reply]
I can certainly do that; just to double check that you're aware that the README.md isn't the documentation itself (anymore)? It's fine if you missed it, I might need to make the Documentation link more prominent since it's definitely where I'd want people going as opposed to the README.md, which is a very rough and quick introduction.

The (perhaps wrong) reasoning for the milestone release notes being where they are is I figured people looking at it might be mostly interested in knowing what's changed in the most recent version. If people are more interested in diving straight into the Getting Started section, then fair enough that's probably what needs highlighting first!

**Reply by emmyarty** [reply]
On a separate note, it's awesome to hear about the project you've got in mind for this; if you hit any limitations where DxSvelte gets in your way instead of facilitating easy development, be sure to let me know - there's a solid chance you won't be alone with it and it would be a great way to make sure this grows into a strong tool fit for real-world use cases

**Reply by baseplate-admin** [reply]
Apologies for yesterday ( i was overworked )..

>  just to double check that you're aware that the README.md isn't the documentation itself (anymore)

Yep I saw this change... That was why i was confused. Some repositories ( [`django-treenode`](https://github.com/fabiocaccamo/django-treenode), [`python-benedict`](https://github.com/fabiocaccamo/python-benedict) ) use `README.md` for their documentation while others ( [`django-rest-framework`](https://www.django-rest-framework.org/community/release-notes/) ) use their docs to keep track of changes..  ( I personally would like to see the changes in the docs. Keeps things organized )

Take a look at [`superforms`](https://github.com/ciscoheat/sveltekit-superforms). [It has a discussion based release](https://github.com/ciscoheat/sveltekit-superforms/discussions/98).

Which in turn looks like this :

![image](https://user-images.githubusercontent.com/61817579/232646966-62f74860-cd00-4970-8fd5-b3d00e90c964.png)


> I might need to make the Documentation link more prominent since it's definitely where I'd want people going as opposed to the README.md, which is a very rough and quick introduction.

Then perhaps you should add your site url so it shows up here ?
![image](https://user-images.githubusercontent.com/61817579/232630890-92f73e2e-bc95-4736-8dc3-d28533762aa1.png)


> The (perhaps wrong) reasoning for the milestone release notes being where they are is I figured people looking at it might be mostly interested in knowing what's changed in the most recent version.


Then perhaps you can use github release with github actions to deploy your packages? Usually when people star your repository and you publish a release, I can look at the changes from `For You` tab.

![image](https://user-images.githubusercontent.com/61817579/232629450-184b6405-0692-4a6f-b1dc-d2329ab5add7.png)

> If people are more interested in diving straight into the Getting Started section, then fair enough that's probably what needs highlighting first

In my opinion i would like to see the features before i would be interested to know what the new version ships ( could be a me problem here )

> On a separate note, it's awesome to hear about the project you've got in mind for this;

Thank you. The project is [`coreproject`](https://github.com/baseplate-admin/CoreProject/).. [which is going through an UI overhaul](https://v3.coreproject.moe/myanimelist/1)

> if you hit any limitations where DxSvelte gets in your way instead of facilitating easy development, be sure to let me know

Yes about this. Would you support packages written for `svelte-kit` ? I specifically would like to see support for UI frameworks like [`skeleton`](https://github.com/skeletonlabs/skeleton)

> it would be a great way to make sure this grows into a strong tool fit for real-world use cases

Yep. Since the entire svelte ecosystem is moving towards `svelte-kit`, I think it would be better if you supported the SSR logic behind `svelte-kit`, so packages written for `svelte-kit` works seamlessly in this repo.

Thank you for reading this long post

**Reply by emmyarty** [reply]
>Take a look at [superforms](https://github.com/ciscoheat/sveltekit-superforms). https://github.com/ciscoheat/sveltekit-superforms/discussions/98.

Thanks; I quite like this approach, that'll be how we do it going forwards.

>Then perhaps you should add your site url so it shows up here ?

Good shout; not sure why adding it to the package slipped my mind, that'll be included going forwards.

>Then perhaps you can use github release with github actions to deploy your packages? Usually when people star your repository and you publish a release, I can look at the changes from For You tab.

Happy to use releases, not sure I'll need actions per se.

>In my opinion i would like to see the features before i would be interested to know what the new version ships ( could be a me problem here )

That's valuable to know; if I'm using releases & discussions to highlight feature changes then omitted them from a third place is no big deal imho. We can do that for sure.

>Thank you. The project is [coreproject](https://github.com/baseplate-admin/CoreProject/).. [which is going through an UI overhaul](https://v3.coreproject.moe/myanimelist/1)

Hey, that looks pretty slick!

>Yes about this. Would you support packages written for svelte-kit ? I specifically would like to see support for UI frameworks like [skeleton](https://github.com/skeletonlabs/skeleton)

This one's a tricky one to answer. SvelteKit, not exactly; SvelteKit brings a lot of utility functions to the table unique to SvelteKit that have no mirror in DxSvelte, _however_ Svelte packages tend to work great. For example, the documentation site uses a package called [svelte-markdown](https://www.npmjs.com/package/svelte-markdown) under the hood to render the markdown sent within the JSON payloads.

Specifically regarding UI frameworks, PostCSS (and thereby Tailwind etc.) support is something I'm working on getting right. It's not over the line yet, so for now you're best off developing apps using their CDN links and importing the full stylesheets, but it's not due to remain a neglected feature (you can find early references to the functionality in the source code for DxSvelte atm).

>Yep. Since the entire svelte ecosystem is moving towards svelte-kit, I think it would be better if you supported the SSR logic behind svelte-kit, so packages written for svelte-kit works seamlessly in this repo.

There are many irreconcilable differences between SvelteKit and Django, sadly, that will make true 1:1 SvelteKit compatibility impossible. Take, for instance, **+page.server.ts** files.  There's simply no equivalent to this in Django; your server logic happens purely within Python, and to truly bridge that gap would be to fragment the server itself.

The good news though is that many packages advertising SvelteKit compatibility are truthfully only dependent on Svelte itself, and are advertising SvelteKit for clarity. Skeleton UI's one example of this; unless I've missed something in the docs, it'll work just great in DxSvelte once PostCSS support is baked in.

>Thank you for reading this long post

Thanks for taking the time to share it!

**Reply by baseplate-admin** [reply]
Alright that about sums up my suggestions for this repository at this point. I will be recommending this repo to users in `svelte` discord. Like this : 
![image](https://user-images.githubusercontent.com/61817579/232997933-de5bd89d-fff1-40db-9bd5-3875c31e50e9.png)

I hope this project becomes a success :D

**Reply by emmyarty** [reply]
Haha awesome man, thanks for the plug 🥳
